bagelbids/cli.py

```diff
@@ -196,29 +232,30 @@ def bids(
         # For some reason .get_sessions() doesn't always follow alphanumeric order
         # By default (without sorting) the session lists look like ["02", "01"] per subject
         for session in sorted(layout.get_sessions(subject=bids_sub_id)):
-            image_list = []
-            for bids_file in layout.get(
-                subject=bids_sub_id,
+            create_session(
```

Comment:
your function `create_session` changes (or mutates) the `session_list` parameter even when you call it (as you do here) without ever assigning its return value. That's called a side-effect (i.e. the function does things other than just return some transform of your inputs) and those are generally not a good idea because it can get confusing quickly and have unintended (side :) ) effects.

I think a better approach would be to say: "hello function, please create a (new) list for me and put in it as many AcquisitionObjects as you can find for this subject/session". And then you figure out what to do with the list you get back from the function. For example: create a fresh Session object and use the returned list as the `hasAcquisiton` attribute.